script.js

```diff
@@ -136,18 +206,40 @@ my_widget_script =
         //LA calls this method with b_suppress_message and relies on your code to communicate issues to the user
         //Returning an empty array [] or NULL equals no error
         //TO DO write code specific to your form
-     
         return this.parent_class.is_valid(b_suppress_message);
       },
-     
+
       is_edited:function () {
         //should return true if the form has been edited since it was loaded or since reset_edited was called
         return this.parent_class.is_edited();
       },
-     
+
       reset_edited:function () {
         //typically called have a save
         //TO DO write code specific to your form
         return this.parent_class.reset_edited();
-      }
-    }
+      },
+
+      // Enable all records if the first record fields are correctly filled
+      enable_all_records: function() {
+        console.log('**** In enable_all_records ****')
+        $('#the_form tbody tr:not(.initialRow) input').prop('disabled', false)
+      },
+
+      // Change all record when amount or moles of first record is changed
+      change_record_value: function() {
+        console.log('**** change_record_value ****')
+        $('#the_form tbody tr:not(.initialRow)').each(function() {
+          var moles = $('input[name^=moles]', this)
+          var amount = $('input[name^=amount]', this)
+          var fw = $('input[name^=fw]', this)
+          var equiv = $('input[name^=equivalents]', this)
+          if (fw.val()) {
```

Comment:
**when initialising rows greater than 1**  - 
if fw is present then calculate equiv and moles, when amnt is entered 
in this case dependency is on fw and amnt and moles, equiv can be blank as they will be auto calculated.

**when amnt is changed for rows greater than 1**
recalculate moles and equiv
In this case there is already valid values for all fields so there is no such checks needed.

Reply:
I'm confused what that response is about. The context here is not initialising rows greater than 1, nor amnt changed for rows greater than 1. The context here is updating all non-initial rows due to a change in the first row. If the subsequent row is only partially filled in (i.e. fw but no equiv) then this will do strange things.

Reply:
I see now that you didn't know what context this code was in. Click the filename to view the context, e.g. 
![image](https://user-images.githubusercontent.com/78827/55523187-b7029a00-56d3-11e9-9124-c0b33486b422.png)